ci/test_cpp_memcheck.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+. /opt/conda/etc/profile.d/conda.sh
+
+rapids-logger "Generate C++ testing dependencies"
+rapids-dependency-file-generator \
+  --output conda \
+  --file_key test_cpp \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch)" | tee env.yaml
+
+rapids-mamba-retry env create --force -f env.yaml -n test
+
+# Temporarily allow unbound variables for conda activation.
+set +u
+conda activate test
+set -u
+
+CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
+RAPIDS_TESTS_DIR=${RAPIDS_TESTS_DIR:-"${PWD}/test-results"}/
+mkdir -p "${RAPIDS_TESTS_DIR}"
+
+rapids-print-env
+
+rapids-mamba-retry install \
+  --channel "${CPP_CHANNEL}" \
+  libcudf libcudf_kafka libcudf-tests
+
+rapids-logger "Check GPU usage"
+nvidia-smi
```

Comment:
It looks like a lot of this code is the same as what's present in the `test_cpp.sh` script.

I think we can DRY this up a bit.

Take a look at the Python test scripts for `cudf` below, which make use of a `test_python_common.sh` script to remove duplication like this.

- https://github.com/rapidsai/cudf/blob/branch-23.04/ci/test_python_common.sh
- https://github.com/rapidsai/cudf/blob/branch-23.04/ci/test_python_cudf.sh
- https://github.com/rapidsai/cudf/blob/branch-23.04/ci/test_python_other.sh

Reply:
Done. Let me know what you think.